infrastructure/modules/azurerm_apim/main.tf

```diff
@@ -0,0 +1,90 @@
+# API management
+
+## Create and configure the API management service
+
+# resource "azurerm_api_management" "azurerm_apim" {
+#   count = "${var.create ? 1 : 0}"
+
+#   name                      = "${local.azurerm_apim_name}"
+#   location                  = "${var.location}"
+#   resource_group_name       = "${var.resource_group_name}"
+#   publisher_name            = "${var.publisher_name}"
+#   publisher_email           = "${var.publisher_email}"
+#   notification_sender_email = "${var.notification_sender_email}"
+
+#   sku {
+#     name     = "${var.sku_name}"
+#     capacity = "${var.sku_capacity}"
+#   }
+
+#   hostname_configuration {}
+# }
+
+module "azurerm_api_management" {
+  source              = "../terraform-azurerm-resource"
+  api_version         = "2019-01-01"
+  type                = "Microsoft.ApiManagement/service/apis"
+  name                = "${local.azurerm_apim_name}"
+  resource_group_name = "${var.resource_group_name}"
+  location            = "${var.location}"
+
+  properties {
+    publisherEmail = "${var.publisher_email}"
+    publisherName  = "${var.publisher_name}"
+    virtualNetworkType = "internal"
+    virtualNetworkConfiguration = {
+      subnetResourceId = "/subscriptions/ec285037-c673-4f58-b594-d7c480da4e8b/resourceGroups/agid-rg-test/providers/Microsoft.Network/virtualNetworks/agid-redis-vnet-test/subnets/default"
```

Comment:
is this a draft ?

Reply:
Yes, just to issue the "terraform plan"  and see the json